### PR TITLE
Add hello-world YAML

### PR DESCRIPTION
**Pull request by parasense**
Add the yaml representation of the hello-world tekton bundle.

This information was fetched from the existing material:
* podman pull quay.io/hacbs-release/hello-world:0.3-alpine
* podman inspect quay.io/hacbs-release/hello-world:0.3-alpine

Signed-off-by: Jon Disnard <jdisnard@redhat.com>

### CODE REVIEW

**Comment by parasense** [comment]
This is a trivial squash merge. Pushing it through.

**Review by johnbieren — APPROVED** [review]

**Review by davidmogar — CHANGES_REQUESTED** [review]
You are missing `---` between resources.

**Comment by davidmogar** [comment]
In addition to my comment, would it make sense to have different files per resource? Maybe not. @johnbieren @parasense opinions?

**Comment by johnbieren** [comment]
> In addition to my comment, would it make sense to have different files per resource? Maybe not. @johnbieren @parasense opinions?

I don't have a strong opinion. If it is a hello-world example that is going to stay very simple like this, I think it is fine. If it will eventually evolve, I would vote for splitting it. Good catch on the `---` though

File: hello-world/hello-world.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  creationTimestamp: null
+  name: hello-world
+spec:
+  tasks:
+  - name: echo-hello-world
+    params:
+    - name: text
+      value: Hello, World!
+    taskRef:
+      bundle: quay.io/hacbs-release/hello-world:0.3-alpine
```
Comment (scoheb) [inline hello-world/hello-world.yaml:31]: sorry. there is something I do not understand. from this line I read that this taskRef essentially refers to itself?

my understand is that this will eventually cause a bundle to be built yet it will refer to itself?

Comment (davidmogar) [reply hello-world/hello-world.yaml:31]: That was necessary in M3 as both the pipeline and the task are in the same bundle. Not sure if there’s a proper way to do this. This is also partly why I preferred bundles to contain tasks exclusively. It would remove this references.

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+---
```
Comment (davidmogar) [inline .github/workflows/lint.yaml:1]: This file should be probably in a different PR as it’s unrelated to this one.

File: .github/workflows/lint.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  workflow_dispatch:
+
+jobs:
+  yamllint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install yamllint
+        run: sudo apt-get install yamllint
```
Comment (davidmogar) [inline .github/workflows/lint.yaml:19]: Have you checked if there are existing actions for yaml linting? There has to be something out there.

**Comment by parasense** [comment]
> You are missing `---` between resources.

Fixed.
Created a yaml linting action to catch this kind of stuff.
Had to reformat the file to address things like indentation, and `---` separators, etc...

File: .github/workflows/lint.yaml (hunk 3 above)
Comment (parasense) [reply .github/workflows/lint.yaml:19]: I wrote this one myself, it was my first Github action, and was based on GitHub's helloworld action.
I'm happy to use something else, I'm sure there are better linters.
I'll take a look.

File: .github/workflows/lint.yaml (hunk 2 above)
Comment (parasense) [reply .github/workflows/lint.yaml:1]: Fair enough.
I'll move linting to another PR.

**Review by davidmogar — CHANGES_REQUESTED** [review]

File: hello-world/hello-world.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      command:
+        - echo
+      image: alpine
+      name: echo
+      resources: {}
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  creationTimestamp: null
```
[the remark this answers is not in the document]
Comment (parasense) [reply hello-world/hello-world.yaml]: Removed these empty or null nodes

File: .github/workflows/lint.yaml (hunk 3 above)
Comment (parasense) [reply .github/workflows/lint.yaml]: Okay, so I've already moved the YAML lint stuff out to another PR, so by now this is a moot topic. However, just to add closure to the thread, I did look again and I found several GitHub actions that implement YAML linting... just like mine.
https://github.com/marketplace?type=actions&query=yamllint

Moving this conversation to that PR, please.
https://github.com/release-service/release-bundles/pull/2

**Review by davidmogar — APPROVED** [review]
